### ISSUE TITLE
export(NodePath) references to the node and not the path

### ISSUE THREAD

**Issue by yaelatletl** [godotengine/godot#36548]
**Godot version:**
3.2 and 3.2.1 rc1 
**OS/device including version:**
Windows 10 64 bit
**Issue description:**
An example: 

tool
extends Position3D

```
export(NodePath) var target 
var from : Spatial

func _ready():
	print(target)
	from = get_node(target)
	print(from)

```

will print this in-editor:
../../../Skeleton2/LThigh
[BoneAttachment:66373]
../../../Skeleton2/LCalf/Position3D
[Position3D:66376]
../../../Skeleton2/LFoot
[BoneAttachment:66377]
but will print this in-game:

[Object:null]
[Object:null]
[Object:null]

I also got it to throw an error: 

> Assigned type (Object) doesn't match the function argument's type (NodePath).  

with the same above code, but I've been unable to recreate it. This issue was found when trying to find a workaround to #34415 

I've included the project with these files, the bug is present in src/Player/Player.tscn

**Steps to reproduce:**
Using the above code and set the export value to some node

**Minimal reproduction project:**

[bug_mannequin.zip](https://github.com/godotengine/godot/files/4253474/bug_mannequin.zip)

**Comment by bojidar-bg** [comment]
In the example project, you have Editable Children enabled on the main scene, where the "target" property of the bone attachments is overridden to be `""`. Probably it happened due to some resurfacing of #5997.

E.g. `src/Main/Game.tscn`, line 46:
```ini
[node name="LCalfCopy" parent="Player/Mannequiny/root/Skeleton/BoneAttachment3" index="0"]
target = NodePath("")
```

**Comment by yaelatletl** [comment]
You are right. It seems that both of those issues were caused by the same issue. I'll close this one.

**Comment by yaelatletl** [comment]
Or well, perhaps you know better, wheter to re-open that one or move the conversation here

**Comment by KoBeWi** [comment]
Is this still valid?
An updated MRP (for Godot 4.4) would be useful, because the original one is very outdated.
